libcxx/include/__memory/shared_ptr.h

```diff
@@ -1137,7 +1137,8 @@ struct __unbounded_array_control_block<_Tp[], _Alloc> : __shared_weak_count
         __alloc_.~_Alloc();
         size_t __size = __unbounded_array_control_block::__bytes_for(__count_);
         _AlignedStorage* __storage = reinterpret_cast<_AlignedStorage*>(this);
-        allocator_traits<_StorageAlloc>::deallocate(__tmp, _PointerTraits::pointer_to(*__storage), __size);
+        allocator_traits<_StorageAlloc>::deallocate(
+            __tmp, _PointerTraits::pointer_to(*__storage), __size / sizeof(_AlignedStorage));
```

Comment:
Is this not always `__count_ + 1`? i.e. the number of elements in the array + the size of the control block at the beginning? I guess I'm kinda confused with my own code now.

Reply:
If `alignof(size_t) < sizeof(size_t)`, it will not be `__count_ + 1`.